tinygrad/codegen/uops.py

```diff
@@ -154,22 +154,26 @@ def compile(u: UOp, name:Optional[str]=None) -> UPat:
 T = TypeVar("T")
 def __unmatch(m1:Union[T, Set[T]], m2:T) -> bool: return m2 not in m1 if isinstance(m1, set) else m2 != m1
 
-def _match(uop:UOp, pat:UPat, store:Dict[str, UOp]) -> bool:
-  if pat.name is not None and store.setdefault(pat.name, uop) is not uop: return False
-  if pat.arg is not None and __unmatch(pat.arg, uop.arg): return False
-  if pat.dtype is not None and uop.dtype is not None and __unmatch(pat.dtype, uop.dtype): return False
-  if pat.op is not None and __unmatch(pat.op, uop.op): return False
-  if pat.src is None: return True
+def __match(uop:UOp, pat:UPat, store:Dict[str, UOp]) -> List[Dict[str, UOp]]:
+  stores, res = [store.copy()], []
```

Comment:
why do you need to copy stores regardless of the commutative branch?
https://github.com/tinygrad/tinygrad/commit/b8a994ab91815b606e284a46c58b3ccea6fdfc0a

Reply:
oh thats good. I think res is needed however